### ISSUE TITLE
500 Page Errors on live site due to login

### ISSUE THREAD

**Issue by chums923** [openhatch/oh-mainline#1579]
### Issue

I've been getting this `500` error on the live site recently. This is a screenshot from when I just accessed the site (using Mac OS 10.7.5 and Google Chrome browser).

![screen shot 2015-04-09 at 12 30 49 pm](https://cloud.githubusercontent.com/assets/2695564/7071562/16d1306e-deb4-11e4-96bf-5fde17be38f0.png)

As you can see, I did it in two tabs just as a sanity check. Same result.

### Replicating this bug

To replicate this bug, you must go to the live site, then login. I've only tested this using the Google login. Once logged in, you're taken to https://openhatch.org/# and you're given a `500` error. Here's a screenshot of that.

![screen shot 2015-04-09 at 12 35 19 pm](https://cloud.githubusercontent.com/assets/2695564/7071692/e38329fa-deb4-11e4-85ae-16a051d2edab.png)

Now, I guess you're 'logged in'. Now, if you navigate to the home page (even if you browse away and come back), you're not only given a `500` error, but the site also makes it seem like you're not logged in. It still has the Log in / Sign up button in the navbar. This is the first screenshot. 

### Cause

Not sure of the exact cause just yet, but I wanted to post this issue to see if anyone had any insights before I start poking around.

**Comment by shaunagm** [comment]
Yeah, I've been getting it too.  I was hoping the fix for the 500 errors you were getting would solve this problem, but it doesn't seem to have.  @paulproteus

**Comment by shaunagm** [comment]
I'm pretty sure your recent PR fixes this.  Can you clear your cache, or open in a new/private browser?  I'm still getting 500 errors too when I open in Mozilla non-privately (how I usually browe) but get a live site in Chrome and when I open a private window.

**Comment by shaunagm** [comment]
I just tried logging into the site on Chrome and the error page appears.  So I think it's actually related to being logged in.  @paulproteus, thoughts?

**Comment by ehashman** [comment]
Yeah, this is weird. I logged in to the site for the missions to help someone today on IRC without any issue, but when I log in and only go to openhatch.org, I get this error (and interestingly, you'll note the top bar does not indicate that you are logged in). When I navigate to any other part of the site, I see that I'm logged in and it loads with no issues.

This is something that needs to be fixed ASAP, I'll take a look at server logs very soon.

**Comment by chums923** [comment]
Great, I'll be interested to know how this turns out. Also, just so you know I had exactly the same experience. I can navigate anywhere on the site except for the home page. When I logged out, though, I could navigate to the home page just fine.

**Comment by moijes12** [comment]
I am facing this too. 
OS: Ubuntu 14.04 LTS on a Virtual box VM
Browser: Firefox 36.0.4

**Comment by ehashman** [comment]
FWIW I am no longer encountering this bug; perhaps one of the recent PRs/deploys fixed it. Can anyone else reproduce?

**Comment by shaunagm** [comment]
@paulproteus - this is fixed, right?  iirc the cause was unknown, but restarting the production removed the unwanted behavior.

**Comment by paulproteus** [comment]
TL;DR: Yes, this is safe to close.

Details:


The cause was:

The deploy script was only partially reloading the code that the server was
using. A partial code reload results in untold insanity, which resulted in
these very mysterious crashes. For more information about this sort of
insanity, please see:
http://foo.ca/wp/chick-tract-satire/who-will-be-eaten-first/


The short-term fix was:

Restart the "uwsgi" daemon on linode.openhatch.org, I did this via: sudo
/etc/init.d/uwsgi restart

on linode.openhatch.org, while logged in as paulproteus.


The near-term fix is:

I adjusted the mysite/scripts/deploy script so that it does "sudo
/etc/init.d/uwsgi restart". (I also adjusted /etc/sudoers on
linode.openhatch.org so that the "deploy" user can run this command as
root.)

This way, the existing mysite/scripts/deploy workflow works as people
expect -- Shauna, Elana, and Carol, and others with the ability to push
code, you should know that because I made this change, you can ignore the
"short-term fix" I listed above. You can run the normal deploy script and
expect a deploy to work properly.


The long-term fix is:

Switch to Heroku, so that these sorts of problems won't happen.

**Comment by ehashman** [comment]
Nice. Good information for the future.

**Comment by paulproteus** [comment]
:heart_decoration: to all who commented here!
